File: Wikipedia/Code/WMFWelcomeLanguageViewController.m
#import "WMFWelcomeLanguageViewController_Testing.h"
#import "Wikipedia-Swift.h"

@interface WMFWelcomeLanguageViewController ()

@property (strong, nonatomic) IBOutlet WelcomeLanguagesAnimationView *animationView;

@end

@implementation WMFWelcomeLanguageViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.animationView.backgroundColor = [UIColor clearColor];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if([segue.destinationViewController isKindOfClass:[WMFWelcomePanelViewController class]]){
        [((WMFWelcomePanelViewController*)segue.destinationViewController) useLanguagesConfiguration];
    }
}

- (void)viewDidAppear:(BOOL)animated {
    BOOL shouldAnimate = !self.hasAlreadyFaded;
    [super viewDidAppear:animated];
    if (shouldAnimate) {
        [self.animationView beginAnimations];
    }
}

@end

File: Wikipedia/Code/WMFWelcomeLanguageViewController.swift

class WMFWelcomeLanguageViewController: WMFWelcomeFadeInAndUpOnceViewController {
    
    @IBOutlet var animationView:WelcomeLanguagesAnimationView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        animationView.backgroundColor = UIColor.clearColor()
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if(segue.destinationViewController.isKindOfClass(WMFWelcomePanelViewController)){
            let panelVC = segue.destinationViewController as! WMFWelcomePanelViewController
            panelVC.useLanguagesConfiguration()
        }
    }
    
    override func viewDidAppear(animated: Bool) {
        let shouldAnimate = !hasAlreadyFaded
        super.viewDidAppear(animated)
        if(shouldAnimate){
            animationView.beginAnimations()
        }
    }
}
